### PR TITLE
[Merged by Bors] - feat: port Topology.UniformSpace.UniformEmbedding

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+open Filter Function Set Uniformity Topology
+
+section
+
+universe u v w
+variable {α : Type u} {β : Type v} {γ : Type w} [UniformSpace α] [UniformSpace β] [UniformSpace γ]
+
+/-- A map `f : α → β` between uniform spaces is called *uniform inducing* if the uniformity filter
+on `α` is the pullback of the uniformity filter on `β` under `prod.map f f`. If `α` is a separated
+space, then this implies that `f` is injective, hence it is a `UniformEmbedding`. -/
```
Comment (dupuisf) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:31]:
```suggestion
/-- A map `f : α → β` between uniform spaces is called *uniform inducing* if the uniformity filter
on `α` is the pullback of the uniformity filter on `β` under `Prod.map f f`. If `α` is a separated
space, then this implies that `f` is injective, hence it is a `UniformEmbedding`. -/
```

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+#align embedding.to_uniform_embedding Embedding.to_uniformEmbedding
+
+section UniformExtension
+
+variable {α : Type _} {β : Type _} {γ : Type _} [UniformSpace α] [UniformSpace β] [UniformSpace γ]
+  {e : β → α} (h_e : UniformInducing e) (h_dense : DenseRange e) {f : β → γ}
+  (h_f : UniformContinuous f)
+
+-- mathport name: exprψ
+local notation "ψ" => DenseInducing.extend (UniformInducing.denseInducing h_e h_dense) f
```
Comment (dupuisf) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:401]:
```suggestion
local notation "ψ" => DenseInducing.extend (UniformInducing.denseInducing h_e h_dense) f
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**